metagraph/src/graph/alignment/aligner_aggregator.hpp

```diff
@@ -4,258 +4,229 @@
+    return true;
+}
 
-    size_t k = graph_.get_k();
-    score_t score = chain.get_score();
-
-    bool called = false;
-    for (auto it = begin; it != end; ++it) {
-        if (it->get_offset())
-            continue;
+template <class AlignmentCompare>
+inline auto AlignmentAggregator<AlignmentCompare>
+::get_min_path_score(const Vector<Column> &labels) const -> score_t {
+    score_t global_min = get_max_path_score() * config_.rel_score_cutoff;
 
-        const char *next_begin = it->get_query().data();
+    if (labels.empty())
+        return std::max(global_min, get_min_path_score());
 
-        assert(chain_begin - chain.get_clipping() == next_begin - it->get_clipping());
-        assert(it->get_orientation() == chain.get_orientation());
+    score_t min_score = std::numeric_limits<score_t>::max();
+    for (Column label : labels) {
+        if (min_score < global_min)
+            break;
 
-        const char *next_end = next_begin + it->get_query().size();
+        min_score = std::min(min_score, get_min_path_score(label));
+    }
 
-        if (next_begin <= chain_begin || next_end == chain_end)
-            continue;
+    return std::max(global_min, min_score);
+}
 
-        Alignment aln(*it);
+template <class AlignmentCompare>
+inline auto AlignmentAggregator<AlignmentCompare>
+::get_max_path_score(const Vector<Column> &labels) const -> score_t {
+    if (labels.empty())
+        return get_max_path_score();
+
+    score_t max_score = std::numeric_limits<score_t>::min();
+    for (Column label : labels) {
+        max_score = std::max(max_score, get_max_path_score(label));
+    }
 
-        if (next_begin >= chain_end) {
-            // no overlap
-            aln.insert_gap_prefix(next_begin - chain_end, graph_, config_);
+    return max_score;
+}
 
-        } else {
-            // trim, then fill in dummy nodes
-            assert(chain.get_end_clipping());
+template <class AlignmentCompare>
+inline auto AlignmentAggregator<AlignmentCompare>
+::get_min_path_score(Column label) const -> score_t {
+    auto find = path_queue_.find(label);
+    return find == path_queue_.end() || find->second.size() < config_.num_alternative_paths
+            || config_.post_chain_alignments
+        ? config_.min_path_score
+        : std::max(static_cast<score_t>(find->second.maximum()->get_score()
+                        * config_.rel_score_cutoff),
+                   find->second.minimum()->get_score());
+}
 
-            // first trim front of the incoming alignment
-            size_t overlap = std::min(
-                static_cast<size_t>((chain.get_cigar().data().end() - 2)->second),
-                aln.trim_query_prefix(chain_end - it->get_query().data(), graph_, config_)
-            );
+template <class AlignmentCompare>
+inline auto AlignmentAggregator<AlignmentCompare>
+::get_max_path_score(Column label) const -> score_t {
+    auto find = path_queue_.find(label);
+    return find == path_queue_.end() ? config_.min_path_score
```

Comment:
Wouldn't it make more sense to return `std::numeric_limits<score_t>::min();`?

Reply:
hmm, I guess so. I added a check later on to the extender to set the upper bound to the max of this and `min_path_score`, so here it's redundant.